### PR TITLE
special/bin: climb up directories to find binary module

### PR DESCRIPTION
**Pull request by sveyret**
Issue #436

### CODE REVIEW

**Comment by sveyret** [comment]
Note that the same problem exists for ignoreBinPackages. I am about to update it too, please wait before merging.

**Comment by rumpl** [comment]
It would seem that you need to rebase your branch against master too

**Comment by rumpl** [comment]
Could you add a test too ?

**Comment by sveyret** [comment]
> It would seem that you need to rebase your branch against master too

Are you sure? It says it is already up to date…

**Comment by rumpl** [comment]
Actually it fails because of the changes I made in a dependent build due to #433 

As soon as that build passes I'll merge it and then you can rebase.

Please don't ask what is this dependent build, I'll work on removing it :)

**Comment by rumpl** [comment]
Some tests would be nice yes :)

Thank you for your work, it's appreciated

**Comment by sveyret** [comment]
I added tests and also made some other modifications:

- changed the per-directory .gitignore into a single one into the test directory (they all were the same)
- removed a test which was duplicate

Each of these modification are in a separate commit for if you do not want to take them.

**Comment by rumpl** [comment]
Thanks! I’ll take a look at it closely tomorrow but it looks good from my phone. Cheers

Also, happy halloween if you are into that kind of things :)

**Comment by rumpl** [comment]
Unrelated but I used to work in Annecy, tres belle ville :)

**Comment by sveyret** [comment]
@rumpl Just a reminder as you said you'd have a look at this one today… :smile:

File: test/spec.js [hunk 1]
```diff
@@ -499,19 +499,6 @@ export default [
       },
     },
   },
-  {
```
Comment (rumpl) [inline test/spec.js]: Why did you remove this test?

Comment (sveyret) [reply test/spec.js]: If I understood well, it's doing exactly the same thing as 2 tests above (report unused bin dependencies if ignoreBinPackage equal false).

Comment (rumpl) [reply test/spec.js]: ooooh good catch!

**Comment by rumpl** [comment]
Thanks for all the work you have been doing lately!

**Comment by sveyret** [comment]
You're welcome. Actually, I'm just making it work for my project… :wink:

**Comment by rumpl** [comment]
Need anything else or do you want me to make a release just for you?

**Comment by sveyret** [comment]
Well, I made everything I needed, so yes, if you can make a release, that'd be great!

**Comment by rumpl** [comment]
Lets get #451 in it too
